### PR TITLE
2.x feature/rewarded ads view

### PR DESCRIPTION
**Pull request by Micpol**
Following changes has been made:

- Support for Admob RewardedAds to BaseBannerActivity
- RewaredDialogFragment to inform user with rewards obtainable through watching ads
- Builder to customize RewaredDialogFragment

### CODE REVIEW

File: library/src/main/java/pl/netigen/rewards/LayoutMargins.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    private Integer left;
+    private Integer top;
+    private Integer right;
+    private Integer bottom;
+
+    private LayoutMargins() {
+    }
+
+    void setNewLayoutParamsForView(View view){
+        ConstraintLayout.LayoutParams layoutParams = (ConstraintLayout.LayoutParams) view.getLayoutParams();
```
Comment (WrobelKrakow) [inline library/src/main/java/pl/netigen/rewards/LayoutMargins.java]: ConstraintLayout maybe can be more general ViewGroup.MarginLayoutParams

File: library/src/main/java/pl/netigen/rewards/RewardDialogFragment.java [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+import java.util.List;
+
+import pl.netigen.netigenapi.BaseBannerActivity;
+import pl.netigen.netigenapi.R;
+
+public class RewardDialogFragment extends AppCompatDialogFragment {
+
+    private static final String TAG = "RewardDialogFragment";
+
+    public static double MATCH_PARENT = ViewGroup.LayoutParams.MATCH_PARENT;
```
Comment (WrobelKrakow) [inline library/src/main/java/pl/netigen/rewards/RewardDialogFragment.java:37]: Unused:
public static double MATCH_PARENT
public static double WRAP_CONTENT

File: library/src/main/java/pl/netigen/rewards/RewardParams.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import androidx.annotation.StringRes;
+
+import java.util.List;
+
+class RewardParams {
+
+    @StringRes
+    int buttonPositiveTextResId = 0;
+    @StringRes
+    int rewardDescrpitonTextResId = 0;
```
Comment (WrobelKrakow) [inline library/src/main/java/pl/netigen/rewards/RewardParams.java]: Misspeling:
"rewardDescrpitonTextResId"  should be rewardDescriptionTextResId

**Review by WrobelKrakow — CHANGES_REQUESTED** [review]

File: library/src/main/java/pl/netigen/rewards/LayoutMargins.java (hunk 1 above)
Comment (Micpol) [reply library/src/main/java/pl/netigen/rewards/LayoutMargins.java]: Done

File: library/src/main/java/pl/netigen/rewards/RewardDialogFragment.java (hunk 2 above)
Comment (Micpol) [reply library/src/main/java/pl/netigen/rewards/RewardDialogFragment.java:37]: It is used by library users to correctly set match_parent or wrap_content

File: library/src/main/java/pl/netigen/rewards/RewardParams.java (hunk 3 above)
Comment (Micpol) [reply library/src/main/java/pl/netigen/rewards/RewardParams.java]: Done

**Review by WrobelKrakow — APPROVED** [review]

**Pull request merged**